dcpy/models/lifecycle/ingest.py

```diff
@@ -38,55 +39,92 @@ class ScriptSource(BaseModel, extra="forbid"):
 )
 
 
-class PreprocessingStep(BaseModel):
+class PreprocessingStep(SortedSerializedBase):
     name: str
     args: dict[str, Any] = {}
     # mode allows for certain preprocessing steps only to be run if specified at runtime
     mode: str | None = None
 
 
-class Template(BaseModel, extra="forbid", arbitrary_types_allowed=True):
-    """Definition of a dataset for ingestion/processing/archiving in edm-recipes"""
+class DatasetAttributes(SortedSerializedBase):
+    name: str | None = None
+    description: str | None = None
+    url: str | None = None
+    custom: dict | None = None
 
-    id: str
+    _head_sort_order = ["name", "description", "url"]
+
+
+class ArchivalMetadata(SortedSerializedBase):
+    archival_timestamp: datetime
+    check_timestamps: list[datetime] = []
+    raw_filename: str
     acl: recipes.ValidAclValues
 
-    target_crs: str | None = None
 
-    ## these two fields might merge to "source" or something equivalent at some point
-    ## for now, they are distinct so that they can be worked on separately
-    ## when implemented, "None" should not be valid type
+class Ingestion(SortedSerializedBase):
+    target_crs: str | None = None
     source: Source
     file_format: file.Format
-
+    processing_mode: str | None = None
     processing_steps: list[PreprocessingStep] = []
 
+
+class Column(SortedSerializedBase):
+    id: str
+    data_type: Literal["text", "integer", "decimal", "geometry", "bool", "datetime"]
+    description: str | None = None
+
+
+class Template(BaseModel, extra="forbid"):
```

Comment:
this one will be eventually phased out, correct?

Reply:
No - this is the class that maps to the dataset template definitions

Reply:
Template exists to simply read in and validate the dataset definition

Config is the "computed" config of a dataset. It has the template definition (attributes, instructions to ingest, etc) as well as the details of archival (where to find the archived raw file, timestamp, etc) and any other fields that need to be determined at runtime.

Reply:
I thought It's the `Config` class that's computed, not `Template`?